File: January 2023/26 January 2023/Case-specific Sorting of Strings.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++

class Solution
{
    public:
    string caseSort(string str, int n)
    {
        
        string v = str;
        sort(v.begin(), v.end());
        string ans;
        int big_index = -1;
        
        while(v[++big_index] < 97);
        
        for(int i = 0, small_index = 0; i < n; i++){
            
            ans += str[i] >= 97 ? v[big_index++] : v[small_index++]; 
        }
            
        return ans;
    }
};

//{ Driver Code Starts.

int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;
		string str;
		cin>>str;
		Solution obj;
		cout<<obj.caseSort (str, n)<<endl;
	}
}
// } Driver Code Ends

File: January 2023/26 January 2023/Case-specific Sorting of Strings.py
#User function Template for python3

class Solution:

    def caseSort(self,s,n):
        case = [1 if ele.isupper() else 0 for ele in s]
        s = sorted(list(s))
        upper = s[:sum(case)][::-1]
        lower = s[sum(case):][::-1]
        return ''.join([upper.pop() if x==1 else lower.pop() for x in case])
#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for tt in range(t):
        n=int(input())
        s=str(input())
        print(Solution().caseSort(s,n))
# } Driver Code Ends
